rdmo/accounts/utils.py

```diff
@@ -90,3 +91,10 @@ def get_user_from_db_or_none(username: str, email: str):
     except ObjectDoesNotExist:
         log.error('Retrieval of user "%s" with email "%s" failed, user does not exist', username, email)
         return None
+
+def is_shibboleth_user(user, search_pattern=None) -> bool:
```

Comment:
Ive moved that if condition from `shibboleth_logout` to a utils func. This is what you wanted to check there if `is_shibboleth_user` or not? Makes it more maintainable like this I think...

Reply:
yes, but the check for the user is not needed, and the setting should not be an argument since settings are available everywhere. to much clutter :laughing:

Reply:
ok but I would like to add check a for `user.has_usable_password()` as a default actually, followed by the search. Do you think that would work in all cases?

Reply:
I guess ldap or social account users will not have a password. We could make the function somehow plugable by the settings, and user the regex as default.

Reply:
I would rather revert the changes, if people need another way to specify the logout, they can just create a custom logout view and add it to they `urls.py`.

Reply:
ok, I've reverted the change and added one more test for the `shibboleth_login` view